packages/plugins/plugin-table/src/hooks/useTableIntialisation.ts

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { useEffect } from 'react';
+
+import { create, S, TypedObject } from '@dxos/echo-schema';
+import { PublicKey } from '@dxos/react-client';
+import { getSpace } from '@dxos/react-client/echo';
+import { createView } from '@dxos/schema';
+
+import { type TableType } from '../types';
+
+/**
+ * Initializes a new view for table if one does not exist.
+ * Creates a new schema and registers it with the database.
+ */
+export const useTableIntialisation = (table?: TableType) => {
+  const space = getSpace(table);
+
+  useEffect(() => {
```

Comment:
i think you like this pattern :), but i feel it's clearer to provide a general purpose, testable function and wrap with an effect where it is needed.

```
useEffect(() => {
  if (table && table?.view) {
    initTable(space, table);
  }
}, [space, table])
```